src/vs/code/node/cli.ts

```diff
@@ -319,23 +319,83 @@ export async function main(argv: string[]): Promise<any> {
 			options['stdio'] = 'ignore';
 		}
 
-		const child = spawn(process.execPath, argv.slice(2), options);
+		let child: ChildProcess;
+		if (platform() !== 'darwin') {
+			child = spawn(process.execPath, argv.slice(2), options);
 
-		if (args.wait && waitMarkerFilePath) {
-			return new Promise<void>(resolve => {
+			if (args.wait && waitMarkerFilePath) {
+				return new Promise<void>(resolve => {
 
-				// Complete when process exits
-				child.once('exit', () => resolve(undefined));
+					// Complete when process exits
+					child.once('exit', () => resolve(undefined));
 
-				// Complete when wait marker file is deleted
-				whenDeleted(waitMarkerFilePath!).then(resolve, resolve);
-			}).then(() => {
+					// Complete when wait marker file is deleted
+					whenDeleted(waitMarkerFilePath!).then(resolve, resolve);
+				}).then(() => {
 
-				// Make sure to delete the tmp stdin file if we have any
-				if (stdinFilePath) {
-					unlinkSync(stdinFilePath);
+					// Make sure to delete the tmp stdin file if we have any
+					if (stdinFilePath) {
+						unlinkSync(stdinFilePath);
+					}
+				});
+			}
+		} else {
+			const requiresWait = args['status'] || args['verbose'] || hasReadStdinArg;
+			const envVars: string[] = [];
+			for (const envKey in env) {
+				// Skip this envKey, otherwise we get a malformed argument warning in the console
+				if (envKey === '_') {
+					continue;
 				}
-			});
+				const value = env[envKey];
+				envVars.push('--env');
+				envVars.push(envKey + '=' + value ?? '');
+			}
+			const openArgs: string[] = ['-n'];
+
+			let tmpfile = '';
+			let tmpfileError = '';
+			if (requiresWait) {
+				const time = new Date().getTime();
+				tmpfile = `${tmpdir()}/vscode-wait-transfer-${time}.log`;
+				tmpfileError = `${tmpdir()}/vscode-wait-transfer-error-${time}.log`;
+				writeFileSync(tmpfile, '');
+				writeFileSync(tmpfileError, '');
+				openArgs.push('-W');
+				openArgs.push('--stdout', tmpfile);
+				openArgs.push('--stderr', tmpfileError);
+				argv.push('--wait');
+			}
+			const argsArr: string[] = [];
+			const execPathToUse = args['exec-path'] ?? process.execPath;
+			argsArr.push('-a', execPathToUse);
+			argsArr.push(...envVars, ...openArgs, '--args', ...argv.slice(2));
+			child = spawn('open', argsArr, options);
+
+			if (requiresWait) {
+				const openPromise = (child: ChildProcess) => new Promise<void>((c) => {
+					if (args['verbose'] || args['status']) {
+						const stream = openSync(tmpfile, 'r');
+						const bufferSize = 500;
+						const buffer = Buffer.alloc(bufferSize);
+						const interval = setInterval(() => {
+							const readAmount = readSync(stream, buffer, 0, bufferSize, null);
+							process.stdout.write(buffer.toString(undefined, 0, readAmount));
+						}, 0);
```

Comment:
Here is another issue with the polling:
For the smoothest output when running with `--verbose` or `--status`, setting `0` here works. However, I assume it also wastes a lot of cycles.

Reply:
I'll just set this to `50`.

Reply:
Resolved by using the file watcher impl